### DISCUSSION TITLE
problems with pip3 install

### DISCUSSION THREAD

**Discussion by NilsNijenhuis** [addiejanssen/domoticz-solaredge-modbustcp-plugin#70] [Q&A]
versie on rpi4 :  6.1.0-rpi4-rpi-v8  / bookworm

can't execute the command : sudo pip3 install -r requirements.txt
I got this error:
```
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.

    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.

    For more information visit http://rptl.io/venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
```
Can you help me?

**Comment by looptmetstok** [comment]
I have the same problem. After a reinstall of domoticz on a RasPi 4 (RasPi OS: bookworm 64bits, Domoticz: 2024.2)

How can I fix this?

**Comment by schroot72** [comment]
I made it work by executing 

python3 -m pip config set global.break-system-packages true

before the pip3 command.

**Reply by Peter-Bos** [reply]
That doesn't work for me. If I look at the internet I guess I have to create a virtual environment. Since I'm a newbie in Raspi OS I get confused of that. Is no simpler solution available?
The bad thing is that I have the ModbusTCP working on another Raspberry Pi that is four yours old and apparently has an older version of Raspi OS.

**Comment by bamloeff** [comment]
Had the same problem in Raspberry Pi Os Trixie. 

Use this command:

sudo pip3 install -r requirements --break-system-packages

This should solve the problem with the externaly-managed-environment error.

But it still didn't work on my newly installed Rpi4. I had to remove pymodbus 3.5.4 and replace it with version 2.4.0 in order to let it work with solaredge_modbus 0.7.0

to check which version pymodbus you have: pip3 list | grep pymodbus
sudo pip3 uninstall pymodbus
sudo pip3 install pymodbus===2.4.0 --break-system-packages

Good luck
